requirements/edx/base.in

```diff
@@ -0,0 +1,189 @@
+# Core dependencies for running edx-platform (LMS or Studio)
+#
+# DON'T JUST ADD NEW DEPENDENCIES!!!
+#
+# If you open a pull request that adds a new dependency, you should:
+#   * verify that the dependency has a license compatible with AGPLv3
+#   * confirm that it has no system requirements beyond what we already install
+#   * run "make upgrade" to update the detailed requirements files
+#
+-r ../edx-sandbox/shared.txt  # Dependencies which are also used by code sandboxes
+-r github.in              # Forks and other dependencies not yet on PyPI
+-r local.in               # Packages in edx-platform which have their own setup.py
+-r paver.txt              # Requirements for running paver commands
+
+# Please follow these guidelines whenever you change this file:
+#
+# 1. When adding a new dependency, add a comment to the end of the line
+#    explaining why it is needed.
+# 2. Do not pin exact versions unless absolutely necessary.  When upgrading
+#    an already-pinned package to its latest release, remove the version pin.
+# 3. Do not add indirect dependencies unless a version constraint is needed to
+#    avoid versions which have known problems.
+# 4. If the package is not needed in production, add it to another file such
+#    as development.in or testing.in instead.
+
+analytics-python==1.1.0             # Used for Segment analytics
+attrs==17.2.0                       # Reduces boilerplate code involving class attributes
+Babel==1.3                          # Internationalization utilities
+beautifulsoup4==4.1.3               # Library for extracting data from HTML and XML files
+bleach==1.4                         # Allowed-list-based HTML sanitizing library that escapes or strips markup and attributes
+boto==2.39.0                        # Deprecated version of the AWS SDK; we should stop using this
+boto3==1.4.8                        # Amazon Web Services SDK for Python
+botocore==1.8.17                    # via boto3, s3transfer
+celery==3.1.25                      # Asynchronous task execution library
+defusedxml==0.4.1                   # XML bomb protection for common XML parsers
+Django==1.11.12                     # Web application framework
+django-babel-underscore             # underscore template extractor for django-babel (internationalization utilities)
+django-birdcage                     # CSRF token forwards compatibility for the Django 1.11 upgrade; can be removed after that
+django-config-models==0.1.8         # Configuration models for Django allowing config management with auditing
+django-cors-headers==2.1.0          # Used to allow to configure CORS headers for cross-domain requests
+django-countries==4.6.1             # Country data for Django forms and model fields
+django-fernet-fields                # via edx-enterprise (should be added to its setup.py)
```

Comment:
@jmbowman We had a little discussion about this at https://github.com/edx/edx-enterprise/pull/303/files#r180804786 -- can you comment on whether we need to add it to edx-enterprise and erase it from here, or...?

cc @douglashall @itsjeyd

Reply:
`django-fernet-fields` is the edX-recommended package to use for storing encrypted values in the database. We added it to edx-platform requirements, so that future use cases wouldn't have to.

Reply:
My thinking was that because django-fernet-fields is only used by edx-enterprise, it should be listed as a dependency in the `install_requires` field of its `setup.py` (which is currently derived from the content of `requirements/base.txt`).  With the new pip-tools configuration in edx-platform, this dependency gets auto-detected by pip-compile and stored in the generated requirements files with a comment like "via edx-enterprise".

My concern about not having it listed as a dependency of edx-enterprise is that we're then counting on a line in edx-platform that says "we need this because edx-enterprise uses it"; if edx-enterprise ever changes such that it needs a different version or doesn't need it at all anymore, somebody has to remember to go change the appropriate lines in edx-platform requirements files.  If that information is in the edx-enterprise `setup.py`, it will get automatically updated in edx-platform when `make upgrade` is run to update the edx-enterprise verion.

Note that this PR was reverted because of issues building sandboxes from scratch (we have some bizarre stuff in our local package installations), but I'm actively fixing that and hope to have it merged again in a day or two.